acceptance/tests/apply/augeas/puppet.rb

```diff
@@ -0,0 +1,36 @@
+test_name "Augeas puppet configuration" do
+
+  tag 'risk:medium'
+
+  confine :except, :platform => 'windows'
+
+  teardown do
+    hosts.each do |host|
+      on host, "cat /tmp/puppet.conf.bak > #{host.puppet['confdir']}/puppet.conf && rm /tmp/puppet.conf.bak"
+    end
+  end
+
+  hosts.each do |host|
+    step "Backup the puppet config"
+    on host, "cp #{host.puppet['confdir']}/puppet.conf /tmp/puppet.conf.bak"
+
+    #Frictionless installs of an agent leave the agent without a [main] or [agent] section in the puppet.conf file.
+    #This invalid formatted puppet.conf file is fine for most things, but will break when augeas runs agianst it. 
+    #This test is for testing augues, and shouldn't be concerend if puppet.conf is perfectly formatted. 
+    #So we check to see if there is a main section, and if not we add that and an agent section in.
+    if on(host, "grep '\\[main\\]' #{host.puppet['confdir']}/puppet.conf" , :acceptable_exit_codes => [0, 1]).exit_code == 1
+      on(host, "sed -i '1s/^/\[main\]\\n/' #{host.puppet['confdir']}/puppet.conf")
```

Comment:
I have no idea why this isn't failing on our PE acceptance tests, but yes just running masterless agent I see this is failing. Looks like on OSX sed does not require the -i option. Easy fix.